### PR TITLE
feat: upgrade edx-drf-extensions and pyjwt

### PR DESCRIPTION
**Pull request by iamsobanjaved**
Upgrading edx-drf-extensions which will bring pyjwt latest version with its breaking changes.

[Release notes](https://github.com/edx/edx-drf-extensions/releases/tag/7.0.0)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -62,9 +62,18 @@ develop: test.requirements
...
+COMMON_CONSTRAINTS_TXT=requirements/common_constraints.txt
+.PHONY: $(COMMON_CONSTRAINTS_TXT)
+$(COMMON_CONSTRAINTS_TXT):
+	wget -O "$(@)" https://raw.githubusercontent.com/edx/edx-lint/master/edx_lint/files/common_constraints.txt || touch "$(@)"
+
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
-upgrade: piptools ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+upgrade: piptools $(COMMON_CONSTRAINTS_TXT) ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	# Make sure to compile files after any other files they include!
+	sed 's/pyjwt\[crypto\]==1.7.1//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
```
Comment (awais786) [inline Makefile:73]: These are temporary changes. Once we remove these pins from common constraints we will clean this repo also.

**Review by awais786 — APPROVED** [review]

File: notesapi/v1/permissions.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def has_permission(self, request, view):
             logger.debug("No token found in headers")
             return False
         try:
-            data = jwt.decode(token, settings.CLIENT_SECRET, audience=settings.CLIENT_ID)
+            data = jwt.decode(token, settings.CLIENT_SECRET, algorithms=["HS256"], audience=settings.CLIENT_ID)
```
Comment (robrap) [inline notesapi/v1/permissions.py:40]: I thought we needed to pick up the algorithm from a setting? Also, why are IDAs decoding? I wonder what is special about this?

File: requirements/common_constraints.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# A central location for most common version constraints
```
Comment (robrap) [inline requirements/common_constraints.txt:1]: I’m still not clear on our process, but couldn’t you update this comment with a link to the real common constraints, and an explanation for having a (what I think is temporary) local copy?

**Review by robrap — COMMENTED** [review]
Also, reminder to search for encode too, which you may have already done.

File: notesapi/v1/permissions.py (hunk 2 above)
Comment (robrap) [reply notesapi/v1/permissions.py:40]: Maybe we can try to trust the comment, assume this is fine, and test in Stage (or sandbox)?

**Review by robrap — APPROVED** [review]
Reminders:
- Search code for all pyjwt breaking changes.
- Test in sandbox or stage before pushing to prod.
- Before merging, I’d check in GoCD if notes has other changes on stage that aren’t yet in Prod. If so, I’d deploy those first so you aren’t testing more changes at once than necessary.

File: notesapi/v1/permissions.py (hunk 2 above)
Comment (iamsobanjaved) [reply notesapi/v1/permissions.py]: Picked up the `algorithm` from settings

**Comment by iamsobanjaved** [comment]
> ### Testing considerations
> This needs special attention before deployment and testing on the stage of specifically API endpoints with JWT authentication enabled. Similar PRs in credentials, analytics-api, and course-discovery already deployed on production.
> 

@edx/engage-squad   kindly review and deploy this PR with these testing considerations

**Comment by jmyatt** [comment]
looks like the deployment pipeline has been broken since PR #263 - we're going to seek help sorting that out before moving forward with this PR

**Comment by mikix** [comment]
I have taken over this branch, if you don't mind. I rebased on master and added a new commit that also removes the constraint on these packages, so they don't just immediately get downgraded in the next requirements-bot PR.

If all the tests pass and this looks good, I'll merge on Thursday and do some testing in stage before a deploy.

**Comment by mikix** [comment]
Seems to work fine in stage - can add/edit/delete/search notes just fine, all of which should require jwt auth. Gonna deploy.

File: notesapi/v1/permissions.py (hunk 2 above)
Comment (robrap) [reply notesapi/v1/permissions.py]: Yes. This was simply to bring this to your attention for some possible future, but not to make it an urgent blocking issue.

Comment (robrap) [reply notesapi/v1/permissions.py]: @mikix: How do you feel about me leaving an in-context note on the topic via a comment? See https://github.com/openedx/edx-notes-api/pull/294

**Pull request merged**